### DISCUSSION TITLE
Wrong type inferred when  using .optional() and .default() in conjunction

### DISCUSSION THREAD

**Discussion by klausXR** [colinhacks/zod#2479] [Q&A] [answered]
I have this schema, the `page` key should be an optional number, which defaults to 1, ie, after parsing, there will always be a value there, however, when I use z.infer and pass in the type, the `page` property is marked as optional.

How can I get a type representation of what the result would look like after parsing and defaults have been applied?

```ts
const Schema = z.object({
  page: z.number().positive().optional().default(1)
})

type SchemaType = z.infer<typeof Schema>
// page?: number | undefined;
```

I tried using z.input and z.output, but they don't work either.

Is this intentional or is there another infer helper, which infers the parsed result?

**Comment by scotttrinh** [comment, marked as answer]
I think you just want `default(1)` in this case. `z.input` should show it as `number | undefined` and `z.output` should show it as `number`.

**Reply by klausXR** [reply]
This syntax is a better fit, but my issue was caused by strict mode not being true in tsconfig
